src/browser_action/vitals.js

```diff
@@ -224,53 +233,144 @@
     );
   }
 
-  function addUserTimings(metric) {
+  function addUserTimings(metric, enableUserTiming, enableConsoleTables) {
     switch (metric.name) {
       case "LCP":
-        // LCP has a loadTime/renderTime (startTime), but not a duration.
-        // Could visualize relative to timeOrigin, or from loadTime -> renderTime.
-        // Skip for now.
+        if (metric.attribution && metric.attribution.lcpEntry && metric.attribution.navigationEntry) {
+          const navEntry = metric.attribution.navigationEntry;
+          // Set the start time to the later of the actual start time or the activationStart (for prerender) or 0
+          const startTime = Math.max(navEntry.startTime, navEntry.activationStart) || 0;
+          // Add the performance marks for the Performance Panel
+          if (enableUserTiming) {
+              performance.measure(`[Web Vitals Extension] LCP.timeToFirstByte`, {
+              start: startTime,
+              duration: metric.attribution.timeToFirstByte,
+            });
+            performance.measure(`[Web Vitals Extension] LCP.resourceLoadDelay`, {
+              start: startTime + metric.attribution.timeToFirstByte,
+              duration: metric.attribution.resourceLoadDelay,
+            });
+            performance.measure(`[Web Vitals Extension] LCP.resourceLoadTime`, {
+              start: startTime + metric.attribution.timeToFirstByte + metric.attribution.resourceLoadDelay,
+              duration: metric.attribution.resourceLoadTime,
+            });
+            performance.measure(`[Web Vitals Extension] LCP.elmentRenderDelay`, {
+              duration: metric.attribution.elementRenderDelay,
+              end: metric.value
+            });
+          }
+          // Add a nice console output
+          if (enableConsoleTables) {
+            console.table(
+              [
+                {
+                  'LCP Breakdown': `LCP (${metric.attribution.lcpEntry.element.nodeName})`,
+                  'Time (ms)': metric.value.toFixed(2),
+                },
+                {
+                  'LCP Breakdown': 'timeToFirstByte',
+                  'Time (ms)': metric.attribution.timeToFirstByte.toFixed(2),
+                },
+                {
+                  'LCP Breakdown': 'resourceLoadDelay',
+                  'Time (ms)': metric.attribution.resourceLoadDelay.toFixed(2),
+                },
+                {
+                  'LCP Breakdown': 'resourceLoadTime',
+                  'Time (ms)': metric.attribution.resourceLoadTime.toFixed(2),
+                },
+                {
+                  'LCP Breakdown': 'elementRenderDelay',
+                  'Time (ms)': metric.attribution.elementRenderDelay.toFixed(2),
+                }
+              ]
+            )
```

Comment:
> Convert stringy numbers to numbers (and remove unnecessary sub-ms precision)

Done (though using `Math.round(x,0)` rather than converting to String with toFixed and then back to number)

> Write full metric names

Done, though used Sentence case as per Optimize LCP guide and usual convention.

> Remove LCP

Disagree. I see this table as a summary presentation of the Performance Entry (which also means you can turn the PerformanceEntry option off completely). It's also helpful to give more context (the element name) when there are multiple LCP entries surfaced (which is an annoying thing that happens since we use `reportAllChanges` to ensure the extension UI is updated as soon as possible):

<img width="831" alt="image" src="https://user-images.githubusercontent.com/10931297/232192938-f386f39e-2c6e-4cc5-ab52-36d7fe9683f5.png">

> Refer to metrics as sub-parts [rather than breakdown]

That would probably make sense if we were not including LCP itself which, as per above, I'd like to keep.

It's also referred to as "Breakdown" in the Optimize LCP guide: https://web.dev/optimize-lcp/#lcp-breakdown

Reply:
> Done, though used Sentence case as per Optimize LCP guide and usual convention.

Not convinced the guide got that one right to be honest! These metric names are proper nouns and you'll find TTFB in title case everywhere else: https://www.google.com/search?q=site%3Aweb.dev+%22time+to+first+byte%22

For now let's keep it sentence case for parity in this PR and think about updating the guide.

> > Remove LCP
>
> Disagree.

That's fine, but let's write out the full metric name for consistency with the other rows.

Reply:
Done.

Reply:
Actually I've reverted the metric name back to the acronym as that allows me to fit in the rating:

<img width="1035" alt="image" src="https://user-images.githubusercontent.com/10931297/232603682-5ac86c36-8b76-44e2-8de7-431eae586bd3.png">